polkadot/runtime/common/src/paras_registrar/mod.rs

```diff
@@ -692,6 +862,23 @@ impl<T: Config> OnNewHead for Pallet<T> {
 	}
 }
 
+impl<T: Config> OnCodeUpgrade for Pallet<T> {
+	fn on_code_upgrade(id: ParaId) -> Weight {
+		let mut deposit_info =
+			Deposits::<T>::get(id).expect("The depositor must be known for each parachain; qed");
```

Comment:
That's not a valid proof. Proofs have to be locally reasonable to be worth anything.